### PR TITLE
Adds support for SLES

### PR DESCRIPTION
**Pull request by gmmeyer**
This Pull request will add support for SLES to the chef repo

### CODE REVIEW

File: recipes/_install-linux.rb [hunk 1]
```diff
@@ -89,5 +89,13 @@
       action package_action # default is :install
       allow_downgrade node['datadog']['agent_allow_downgrade']
     end
+  when 'suse'
+    zypper_package 'datadog-agent' do
```
Comment (olivielpeau) [inline recipes/_install-linux.rb]: One more thing: foodcritic fails on one of the resource's attributes. Based on the docs (https://docs.chef.io/resource_zypper_package.html) `allow_downgrade` doesn't exist on this resource. Maybe passing a specific option in `options` instead could work?

**Comment by shinitiandrei** [comment]
@gmmeyer Any updates?

**Comment by gmmeyer** [comment]
@shinitiandrei I totally forgot about this pull request! I'll get it merged! Thanks for the ping on it

**Comment by shinitiandrei** [comment]
@gmmeyer sorry for poking you again, I know how busy our lives can be, so just leaving a reminder here and help everyone in this community :).

Thanks heaps!

File: recipes/repository.rb [hunk 2]
```diff
@@ -100,4 +100,52 @@
...
+      only_if "gpg --dry-run --quiet --with-fingerprint #{new_key_local_path} | grep 'A4C0 B90D 7443 CF6E 4E8A  A341 F106 8E14 E094 22B3'"
+      action :nothing
+    end
+  end
+
+  old_key_local_path = ::File.join(Chef::Config[:file_cache_path], 'DATADOG_RPM_KEY.public')
+  remote_file 'DATADOG_RPM_KEY.public' do
+    path old_key_local_path
+    source node['datadog']['yumrepo_gpgkey']
+    # not_if 'rpm -q gpg-pubkey-4172a230' # (key already imported)
```
[the remark this answers is not in the document]
Comment (gmmeyer) [reply recipes/repository.rb]: I don't remember

[the remark this answers is not in the document]
Comment (gmmeyer) [reply recipes/repository.rb]: I'll uncomment it, I think it was a mistake

File: recipes/_install-linux.rb (hunk 1 above)
Comment (gmmeyer) [reply recipes/_install-linux.rb]: allow downgrade isn't a thing on suse, you're right. I believe chef allows you to downgrade on it without the flag, just by specifying a version.

**Comment by gmmeyer** [comment]
@olivielpeau there is a recipe that allows you to use it on earlier chefs, but it's bad. I tested it out and it doesn't work very well. Are you okay with only supporting chef 13 for this?

**Comment by shinitiandrei** [comment]
Thanks heaps @gmmeyer !!

@olivielpeau Could you take a look into this, please?

File: README.md [hunk 3]
```diff
@@ -39,6 +39,7 @@ Platforms
 * Scientific Linux
 * Ubuntu
 * Windows (requires chef >= 12.0)
+* SUSE (requires chef >= 13.0)
```
Comment (olivielpeau) [inline README.md]: should read `13.3`

File: recipes/repository.rb [hunk 4]
```diff
@@ -84,4 +84,52 @@
...
+  zypper_repository 'datadog' do
+    name 'datadog'
+    description 'datadog'
+    if node['datadog']['agent6']
+      baseurl node['datadog']['agent6_yumrepo_suse']
+    else
+      baseurl node['datadog']['yumrepo_suse']
+    end
+    gpgkey node['datadog']['yumrepo_gpgkey']
+    gpgcheck false
```
Comment (olivielpeau) [inline recipes/repository.rb]: does this disable the signature check on the packages themselves as well?

File: recipes/repository.rb [hunk 5]
```diff
@@ -84,4 +84,52 @@
...
+  # Add YUM repository
+  zypper_repository 'datadog' do
+    name 'datadog'
+    description 'datadog'
+    if node['datadog']['agent6']
+      baseurl node['datadog']['agent6_yumrepo_suse']
+    else
+      baseurl node['datadog']['yumrepo_suse']
+    end
+    gpgkey node['datadog']['yumrepo_gpgkey']
```
Comment (olivielpeau) [inline recipes/repository.rb]: since this key is specified here, does it still need to be imported "manually" above with `remote_file` + `execute` ?

Comment (gmmeyer) [reply recipes/repository.rb]: yea it does, I dunno why but sles demands that for some reason

File: recipes/repository.rb (hunk 4 above)
Comment (gmmeyer) [reply recipes/repository.rb]: I believe it should just do it for the repo

Comment (gmmeyer) [reply recipes/repository.rb]: We don't sign the repo itself so we have to disable it for that

Comment (gmmeyer) [reply recipes/repository.rb]: It's been a while I will double check

Comment (gmmeyer) [reply recipes/repository.rb]: yea it says:

> gpgcheck
> Ruby Type: true, false | Default Value: true

> Determines whether or not to perform a GPG signature check on the repository.

File: recipes/repository.rb (hunk 5 above)
Comment (gmmeyer) [reply recipes/repository.rb]: There might be another way to do it but I dunno what it is

File: attributes/default.rb [hunk 6]
```diff
@@ -56,6 +56,7 @@
 default['datadog']['agent6_aptrepo_dist'] = 'stable'
 # RPMs are only available for RHEL >= 6 (-> use https protocol) and x86_64 arch
 default['datadog']['agent6_yumrepo'] = 'https://yum.datadoghq.com/stable/6/x86_64/'
+default['datadog']['agent6_yumrepo_suse'] = 'https://yum.datadoghq.com/stable/6/x86_64/'
```
Comment (remeh) [inline attributes/default.rb]: I think it should be `https://yum.datadoghq.com/suse/stable/6/x86_64/`?

**Pull request merged**